### PR TITLE
Circuit fixes and updates

### PR DESCRIPTION
**Pull request by saleel**
## Motivation

Add some improvements to the circuits

## Change Summary

- Upgrade to latest version of zkemail 
- Add AssertZeroPadding
- Optimize gender extraction to save ~14k constraints
- Overall circuit improvement and minor bug fixes

## Merge Checklist

_Choose all relevant options below by adding an `x` now or at any time before submitting for review_

- [ ] PR title adheres to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standard
- [ ] PR has a [change set](https://github.com/privacy-scaling-explorations/anon-aadhaar/CHANGELOG.md)
- [ ] PR has been tagged with a change label(s) (i.e. documentation, feature, bug fix, or chore)
- [ ] PR includes documentation if necessary.

## Additional Context

If this is a relatively large or complex change, provide more details here that will help reviewers

### CODE REVIEW

File: packages/circuits/scripts/build.sh [hunk 1]
```diff
@@ -48,7 +48,7 @@ function install_deps() {
 }
 
 function build_circuit() {
-    circom ./src/aadhaar-verifier.circom  --r1cs --wasm -o $BUILD_DIR -l ./node_modules
+    circom ./src/aadhaar-verifier.circom  --r1cs --wasm -o $BUILD_DIR -l ./node_modules -l ../../node_modules
```
Comment (Meyanis95) [inline packages/circuits/scripts/build.sh:51]: Could we add -c compiler option? Could be useful for RapidSnark

File: packages/circuits/src/helpers/extractor.circom [hunk 2]
```diff
@@ -2,12 +2,53 @@ pragma circom 2.1.6;
...
 include "../utils/pack.circom";
 
 
 
+/**
+Aadhar QR code data schema (V2)
+
+V1 Docs - https://uidai.gov.in/images/resource/User_manulal_QR_Code_15032019.pdf
+There are no official spec docs for Aadhaar V2 available publically, but the difference from V1 is:
+  - "V2" is added at the bedinning of the data, before the first delimitter.
```
Comment (Meyanis95) [inline packages/circuits/src/helpers/extractor.circom:17]: Small typo here "beginning"

File: packages/circuits/src/helpers/extractor.circom [hunk 3]
```diff
@@ -2,12 +2,53 @@ pragma circom 2.1.6;
...
+  9. Address > House
+  10. Address > Location
+  11. Address > Pin code
+  12. Address > Post office
+  13. Address > State
+  14. Address > Street
+  15. Address > Sub district
+  16. VTC
+  17. Last 4 digits of mobile number
+  18. The data after 18th 255 till end (-256 ofthe signature) is the photo.
```
Comment (Meyanis95) [inline packages/circuits/src/helpers/extractor.circom:45]: Small typo here "of the"

File: packages/circuits/src/helpers/extractor.circom [hunk 4]
```diff
@@ -2,12 +2,53 @@ pragma circom 2.1.6;
...
+
+- Delimiter is 255.
+- Before first delimiter, there are two bytes representing the version. This should be [86, 50] (V2)
+- From then on, each field is separated by the delimiter. There are total of 16 fields.
+  1 (data after first 255). Email_mobile_present_bit_indicator_value (can be 0 or 1 or 2 or 3): 
+      0: indicates no mobile/email present in secure qr code. 
+      1: indicates only email present in secure qr code. 
+      2: indicates only mobile present in secure qr code 
+      3: indicates both mobile and email present in secure qr code.
+  2. Reference ID (Last 4 digits of aadhar number and timestamp)
```
Comment (Meyanis95) [inline packages/circuits/src/helpers/extractor.circom:29]: I'm still confuse is it aadhaar? or aadhar?

File: packages/circuits/src/helpers/extractor.circom [hunk 5]
```diff
@@ -2,12 +2,53 @@ pragma circom 2.1.6;
...
+  17. Last 4 digits of mobile number
+  18. The data after 18th 255 till end (-256 ofthe signature) is the photo.
+
+- Last 256 bytes is the signature.
+**/
+
+
+
 /// @title ExtractAndPackAsInt
 /// @notice Helper function to exract data at a position to a single int (assumes data is less than 31 bytes)
```
Comment (Meyanis95) [inline packages/circuits/src/helpers/extractor.circom:53]: Small typo "extract"

File: packages/circuits/src/utils/pack.circom [hunk 6]
```diff
@@ -2,80 +2,12 @@ pragma circom 2.1.6;
...
-    out <== sum[length];
-}
-
-
 /// @title DigitBytesToTimestamp
 /// @notice Converts a date string of format YYYYMMDDHHMMSS to a unix time
 /// @notice Each byte is expected to be a ASCII character representing a digit
 /// @notice Assumes the input time is in UTC
+/// @dev Does not work for time before unix epoc (negative timestamps)
+/// @dev Inputs are not sanity checked in this templte (eg: month <= 12?, year >= 1970?)
```
Comment (Meyanis95) [inline packages/circuits/src/utils/pack.circom:10]: Small typo "template"

**Review by Meyanis95 — APPROVED** [review]
LGTM

File: packages/circuits/src/helpers/extractor.circom (hunk 4 above)
Comment (saleel) [reply packages/circuits/src/helpers/extractor.circom]: Aadhaar - this one is a type. I have fixed.

**Pull request merged**